qcore/microtime.py

```diff
@@ -110,27 +107,62 @@ def __init__(self, offset):
         self.offset = int(offset)
 
     def __enter__(self):
-        global _time_offset
-        _time_offset += self.offset
+        global _offset_utime
+        _offset_utime += self.offset
 
     def __exit__(self, typ, value, traceback):
-        global _time_offset
-        _time_offset -= self.offset
+        global _offset_utime
+        _offset_utime -= self.offset
 
 
 def utime():
     """Gets current time in microseconds from the epoch time w/applied offset."""
-    return _time_offset + int(_time() * 1000000)
+    return _offset_utime + int(_time_in_seconds() * SECOND)
 
 
 def true_utime():
     """Gets current time in microseconds from the epoch time."""
-    return int(_time() * 1000000)
+    return int(_time_in_seconds() * SECOND)
+
+
+# ===================================================
+# Conversions to/from PY Date-Time
+# ===================================================
+
+
+def utime_as_datetime(utime, *, drop_subseconds=False, tz=timezone.utc):
```

Comment:
Added it since that behavior has been common in application code, so we can codemod existing conversions without having to update all the unit tests.

Also, I'm not sure if 100% of the current conversions can support maintaining the subsecond.

Reply:
> Added it since that behavior has been common in application code, so we can codemod existing conversions without having to update all the unit tests.

Adding the parameter here would mean that we'd have to maintain it for a long time for backward compatibility reasons, though. I'd rather migrate the internal use cases to the new behavior.

> Uh, also, since the behavior of `format_utime_as_iso_8601` would be very different depending on this, I think it's more useful to keep the support in `format_utime_as_iso_8601()`, and this kind of entails that.

I do agree that the param is more useful in that function, but it could simply call `utime // qcore.SECOND * qcore.SECOND` first to round down.

Reply:
Sure. Let's start with that, then.

Reply:
Okay. Dropped it for this one.